pkg/states/state_manager.go

```diff
@@ -2,20 +2,33 @@ package states
 
 import toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
 
-func Approved(userSignup *toolchainv1alpha1.UserSignup) bool {
+func ManuallyApproved(userSignup *toolchainv1alpha1.UserSignup) bool {
 	return contains(userSignup.Spec.States, toolchainv1alpha1.UserSignupStateApproved)
 }
 
-func SetApproved(userSignup *toolchainv1alpha1.UserSignup, val bool) {
-	setState(userSignup, toolchainv1alpha1.UserSignupStateApproved, val)
-
-	if val {
+// SetManuallyApproved when a user was manually approved by an admin
+// Note: when a user is manually approved, the `UserSignup.Spec.States` contains a single entry: `approved`
+func SetManuallyApproved(userSignup *toolchainv1alpha1.UserSignup, approved bool) {
+	setState(userSignup, toolchainv1alpha1.UserSignupStateApproved, approved)
+	if approved {
 		setState(userSignup, toolchainv1alpha1.UserSignupStateVerificationRequired, false)
 		setState(userSignup, toolchainv1alpha1.UserSignupStateDeactivating, false)
 		setState(userSignup, toolchainv1alpha1.UserSignupStateDeactivated, false)
 	}
 }
 
+func AutomaticallyApproved(userSignup *toolchainv1alpha1.UserSignup) bool {
+	return userSignup.Spec.States == nil
+}
+
+// SetAutomaticallyApproved when a user was automatically approved (verification passed, etc.)
+// Note: when a user is automatically approved, the `UserSignup.Spec.States` is reset
+func SetAutomaticallyApproved(userSignup *toolchainv1alpha1.UserSignup, approved bool) {
```

Comment:
is there any place in production where any of these two functions would be called? I'm not aware of any.

Reply:
I'm in agreement with Matous here.. plus automatic approval is a configuration concern, and i feel it has no place here.

Reply:
I remove this func in 5af9225